### DISCUSSION TITLE
Put a warning message to MOTD and login banner when the secure boot is disabled

### DISCUSSION THREAD

**Discussion by smallorange** [fwupd/fwupd#4759] [Ideas]
Secure boot can be used to protect system security at the firmware level.  If it was intentionally disabled by someone or the malware, it would lead to significant damage to the data. Previously, we had proposed to put the secure boot warning message to the GNOME login dialog and workspace. That could make workstation users know there are some problems with their system.

For most of the server users, their system runs without a GUI interface. We cannot put any warning messages on them. So, MOTD should be a better user interface to notify these users their system is in a dangerous condition. As shown in the figure below, after login via ssh, the user can see the MOTD with our secure boot warning message.
[ssh login](https://drive.google.com/file/d/1Hk38hv5x1MlARtpc6I0izfSyoi5ik2-V/view?usp=sharing)

Also, as shown in the figure below, The message would be set to /etc/issue. The users can immediately see the warning before they try to login through the terminal.
[terminal login](https://drive.google.com/file/d/1JP_Ec0ASEhoaKBbMkiSs78WisMvLHUa9/view?usp=sharing)

Consequently, this warning message can provide an instant warning for those systems without GUI. If the user is aware of this, they should start to consider not login or asking for help. If the function could be added to fwupd, it could be good news for the users.

**Comment by superm1** [comment]
So fwupd already populates the motd with a small blurb if there are updates available.  Is the thought here that it should also populate a second line with security problems?

**Reply by smallorange** [reply]
Yes, the second line with security problems in MOTD.  It could also be placed to /etc/issue which configures the login banner. So, a user who accesses the system locally can quickly know the security status by reading the login banner.

**Reply by superm1** [reply]
Seems fine by me to extend motd.

**Reply by smallorange** [reply]
Great and thank you.
From the point of view of the login banner (/etc/issue), what do you think? I know it may become more complicated to preserve the customized login banner.

**Reply by hughsie** [reply]
I kinda thought it would be implemented by the adding to the existing motd code here: https://github.com/fwupd/fwupd/blob/main/src/fu-engine-helper.c#L48

**Reply by superm1** [reply]
I think start with motd. If it works better for /etc/issue, it's a small amount of code to move around.
